[feature]: (#31) Add new columns in project doctype

Jithendranath-C opened the pull request:
closes [](https://github.com/ParaLogicTech/automotive/issues/31)

Jithendranath-C committed:
[feature]: (#31) Add new columns in project doctype

Jithendranath-C committed:
[feature]: (#31) clean up

SaiFi0102 commented on erpnext/projects/doctype/project/project.py:
```diff
@@ -150,17 +150,20 @@ def set_billing_and_delivery_status(self, update=False, update_modified=False):
+		sales_orders = frappe.get_all("Sales Order", fields=['billing_status', 'delivery_status', 'status', 'skip_delivery_note', 'transaction_date'], filters={
 			"project": self.name, "docstatus": 1
-		})
+		}, order_by="transaction_date desc")
```
`order_by="transaction_date desc, creation desc"` both

SaiFi0102 commented on erpnext/projects/doctype/project/project.py:
```diff
@@ -265,39 +270,47 @@ def get_delivery_status(self, sales_orders, delivery_notes, material_requests):
 				"material_request_type": ["in", ["Purchase", "Material Transfer", "Customer Provided"]],
-			}
+			},
+			order_by = "transaction_date desc",
```
order_by="transaction_date desc, creation desc" both

SaiFi0102 commented on erpnext/projects/doctype/project/project.py:
```diff
@@ -265,39 +270,47 @@ def get_delivery_status(self, sales_orders, delivery_notes, material_requests):
 			from `tabPurchase Receipt Item` i
 			inner join `tabPurchase Receipt` p on p.name = i.parent
-			where p.docstatus = 1 and i.project = %s and i.is_stock_item = 1
+			where p.docstatus = 1 and i.project = %s and i.is_stock_item = 1 order by p.posting_date desc
```
order_by="posting_date desc, creation desc" both and move it to new line for better readability

SaiFi0102 commented on erpnext/projects/doctype/project/project.py:
```diff
@@ -265,39 +270,47 @@ def get_delivery_status(self, sales_orders, delivery_notes, material_requests):
 	def get_procurement_status(self):
 		purchase_orders = frappe.db.sql("""
-			select p.receipt_status, p.status, i.qty, i.received_qty
+			select p.receipt_status, p.status, i.qty, i.received_qty, p.transaction_date
```
order_by="transaction_date desc, creation desc" both and move it to new line for better readability

SaiFi0102 commented on erpnext/projects/doctype/project/project.py:
```diff
@@ -265,39 +270,47 @@ def get_delivery_status(self, sales_orders, delivery_notes, material_requests):
 			from `tabPurchase Order Item` i
 			inner join `tabPurchase Order` p on p.name = i.parent
-			where p.docstatus = 1 and i.project = %s and i.is_stock_item = 1
+			where p.docstatus = 1 and i.project = %s and i.is_stock_item = 1 order by p.transaction_date desc
```
order_by="transaction_date desc, creation desc" both and move it to new line for better readability

SaiFi0102 commented on erpnext/projects/doctype/project/project.py:
```diff
@@ -1273,7 +1290,7 @@ def set_items_and_totals_html_onload(self, sales_data, consumables_data):
 	def get_project_sales_data(self, get_sales_invoice=True):
 		sales_data = frappe._dict()
-		sales_data.material_items, sales_data.part_items, sales_data.lubricant_items = get_material_items(self,
+		sales_data.material_items, sales_data.part_items, sales_data.lubricant_items, sales_data.consumable_items, sales_data.total_discount_amount = get_material_items(self,
```
wrong location for `total_discount_amount`, it should be in `sales_data.totals.total_discount_amount`

SaiFi0102 commented on erpnext/projects/doctype/project/project.py:
```diff
@@ -1405,6 +1422,17 @@ def validate_notification(self, notification_type=None, child_doctype=None, chil
 
+	def set_pending_quotation_total(self, update=False, update_modified=False):
+		total = frappe.db.sql("""
+			SELECT SUM(rounded_total) 
```
- Use `base_rounded_total` (for company currency)
- Get both `base_grand_total` and `base_rounded_total` use `base_rounded_total or base_grand_total` because erpnext makes `base_rounded_total = 0` in certain cases where rounding is disabled

SaiFi0102 commented on erpnext/selling/doctype/quotation/quotation.py:
```diff
@@ -6,6 +6,7 @@
 from frappe.utils import flt, nowdate, getdate, cint, add_days, date_diff
 from frappe import _
 from erpnext.overrides.lead.lead_hooks import get_customer_from_lead
+from erpnext.projects.doctype.project.project import Project
```
remove unnecessary import

Jithendranath-C committed:
[feature]: (#31) PR suggestions

Jithendranath-C committed:
[feature]: (#31) Fix sql queries

SaiFi0102 committed:
Merge remote-tracking branch 'upstream/version-15' into fork/Jithendranath-SE/add_columns_in_reports

SaiFi0102 committed:
fix: PR Review Changes
- patch improvement and missing patches.txt line
- rearrange fields and set missing properties and made visible
- commonified discount field and fix discount calculation
- renamed field pending_quotation_amount
- fix and commonified quotation update project call
- add missing set_pending_quotation_amount in sales order and quotation is lost

SaiFi0102 committed:
fix: PR Review changes and improvements
